.github/workflows/nodejs.yml

```
@@ -15,8 +15,17 @@ jobs:
       matrix:
         node-version: [16.x, 18.x, 20.x]
     steps:
-      - name: Checkout
+      - name: Checkout (PR)
+        if: github.event_name == 'pull_request'
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+
+      # fetch-depth: 0 looks like it has a perf impact, so let's not do that if we don't have to.
+      # see https://github.com/actions/checkout/blob/main/README.md#usage
+      - name: Checkout (Push)
+        if: github.event_name != 'pull_request'
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+        with:
+          fetch-depth: 0 # needed for dirty check
```

```suggestion
          fetch-depth: 2 # needed for dirty check
```
Curious why this is not sufficient :thinking:

---

me too